src/App.tsx

```diff
@@ -5,6 +5,40 @@ import { customersGetter, customersSetter } from './api/CustomerApi';
 import { initialCustomers } from './api/startingValues';
 import ContainerWrapper from './components/ContainerWrapper';
 import Landing from './pages/Landing';
+import { QueryClient, QueryClientProvider, useQueryClient, useMutation } from 'react-query';
+import App from './App';
+import ReactDOM from 'react-dom';
+
+const queryClient = new QueryClient();
+
+ReactDOM.render(
+  <QueryClientProvider client={queryClient}>
+    <App />
+  </QueryClientProvider>,
+  document.getElementById('root')
+);
+
+const addNewCustomer = async (customerData: any) => {
+  const response = await fetch('/api/customers', {
+    method: 'POST',
+    body: JSON.stringify(customerData),
+    headers: {
+      'Content-Type': 'application/json',
+    },
+  });
+
+  if (!response.ok) {
+    throw new Error('Failed to add customer');
+  }
+
+  return response.json();
+};
+const { mutate } = useMutation(addNewCustomer, {
+  onSuccess: () => {
+
+    queryClient.invalidateQueries('customers');
+  },
+});
```

Comment:
This is the only portion of code that may in part be used.
But we wouldn't place this in the app.tsx file.

Reply:
Nvm I see what you mean lol